### PR TITLE
Configure react with appengine

### PR DESCRIPTION
**Pull request by malachibre**
There's an extra step when you want to deploy with react. You have to deploy the front end by running "yarn build" then "gcloud app deploy" while inside the edith-step directory. Then, you deploy the back-end the regular way using "mvn package appengine:deploy" in the root directory. 

When you want run the app on the test server, it looks like you can only test either the front-end or the back-end at one time. To test the front end run "yarn build" then "yarn local" while in the edith-step directory. And again testing the back-end is the same as usual with "mvn package appengine:run" in the root directory.

### CODE REVIEW

File: dispatch.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Rules for dispatch file...
+# Put specific cases at the top.
+dispatch:
+# Route the urls that point to the Java backend's API calls
+ - url: "*/api/v1/*"
+   service: backend
+# Route all other urls to the React.js frontend
+ - url: "*/*"
+   service: default
```
Comment (Prashantneu8848) [inline dispatch.yaml:9]: I think we should add a new line at the end of the file ?

File: edith-step/app.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+runtime: nodejs10
+ 
+resources:
+ cpu: 1
+ memory_gb: 0.5
+ disk_size_gb: 11
```
Comment (Prashantneu8848) [inline edith-step/app.yaml:6]: add a new line ?

File: dispatch.yaml (hunk 1 above)
Comment (malachibre) [reply dispatch.yaml:9]: I added it.

File: edith-step/app.yaml (hunk 2 above)
Comment (malachibre) [reply edith-step/app.yaml:6]: Added.

**Comment by malachibre** [comment]
> Can everything in the edith-step folder be moved to the root folder?

When I moved everything out the edith-step folder, the app wouldn't compile if the app.js file wasn't in the src. So I moved it and all the related files into the src folder, and it worked fine. But it was inconsistent, because it wasn't inside the webapp folder with index.html. If we're okay with the front-end files not being in the webapp, then we can move it back, but if not we can just keep it all the front-end files separate in the edith-step folder.

**Pull request merged**